webv2/table/utilities.go

```diff
@@ -0,0 +1,224 @@
+func GetSpannerTableDDL(spannerTable ddl.CreateTable) string {
+
+	c := ddl.Config{Comments: true, ProtectIds: false}
+
+	ddl := spannerTable.PrintCreateTable(c)
+
+	return ddl
+}
+
+func renameinterleaveTableSchema(interleaveTableSchema []InterleaveTableSchema, table string, columnId string, colName string, newName string) []InterleaveTableSchema {
+
+	tindex := isTablePresent(interleaveTableSchema, table)
+
+	interleaveTableSchema = createinterleaveTableSchema(interleaveTableSchema, table, tindex)
+
+	interleaveTableSchema = renameInterleaveColumn(interleaveTableSchema, table, columnId, colName, newName)
+
+	return interleaveTableSchema
+}
+
+func isTablePresent(interleaveTableSchema []InterleaveTableSchema, table string) int {
+
+	for i := 0; i < len(interleaveTableSchema); i++ {
+		if interleaveTableSchema[i].Table == table {
+			return i
+		}
+	}
+
+	return -1
+}
+
+func createinterleaveTableSchema(interleaveTableSchema []InterleaveTableSchema, table string, tindex int) []InterleaveTableSchema {
+
+	if tindex == -1 {
+		itc := InterleaveTableSchema{}
+		itc.Table = table
+		itc.InterleaveColumnChanges = []InterleaveColumn{}
+
+		interleaveTableSchema = append(interleaveTableSchema, itc)
+	}
+
+	return interleaveTableSchema
+}
+
+func renameInterleaveColumn(interleaveTableSchema []InterleaveTableSchema, table, columnId, colName, newName string) []InterleaveTableSchema {
+
+	tindex := isTablePresent(interleaveTableSchema, table)
+
+	cindex := isColumnPresent(interleaveTableSchema[tindex].InterleaveColumnChanges, columnId)
+
+	interleaveTableSchema = createInterleaveColumn(interleaveTableSchema, tindex, cindex, columnId, colName, newName)
+
+	if tindex != -1 && cindex != -1 {
+		interleaveTableSchema[tindex].InterleaveColumnChanges[cindex].ColumnId = columnId
+		interleaveTableSchema[tindex].InterleaveColumnChanges[cindex].ColumnName = colName
+		interleaveTableSchema[tindex].InterleaveColumnChanges[cindex].UpdateColumnName = newName
+
+	}
+
+	return interleaveTableSchema
+
+}
+
+func createInterleaveColumn(interleaveTableSchema []InterleaveTableSchema, tindex int, cindex int, columnId string, colName string, newName string) []InterleaveTableSchema {
+
+	if cindex == -1 {
+
+		if columnId != "" {
+
+			ic := InterleaveColumn{}
+			ic.ColumnId = columnId
+			ic.ColumnName = colName
+			ic.UpdateColumnName = newName
+
+			interleaveTableSchema[tindex].InterleaveColumnChanges = append(interleaveTableSchema[tindex].InterleaveColumnChanges, ic)
+		}
+	}
+
+	return interleaveTableSchema
+}
+
+func isColumnPresent(interleaveColumn []InterleaveColumn, columnId string) int {
+
+	for i := 0; i < len(interleaveColumn); i++ {
+		if interleaveColumn[i].ColumnId == columnId {
+			return i
+		}
+	}
+
+	return -1
+}
+
+func typeinterleaveTableSchema(interleaveTableSchema []InterleaveTableSchema, table string, columnId string, colName string, previoustype string, updateType string) []InterleaveTableSchema {
```

Comment:
func typeInterleaveTableSchema - function name is not clear about what it does

Reply:
Rename to **updateTypeOfInterleaveTableSchema**